runtime/lua/vim/treesitter/query.lua

```diff
@@ -332,12 +332,28 @@ M.get = memoize('concat-2', function(lang, query_name)
   return M.parse(lang, query_string)
 end, false)
 
+-- Clears the query information cache for a given tuple of [language, query], or all cached query
+-- info if either is not given.
+--
+---@param lang string? Language to use for the query
+---@param query_name string? Name of the query (e.g. "highlights")
+M.clear_cache = function(lang, query_name)
```

Comment:
Can drop the parameters to keep the interface "trivial" as claimed in the PR description.

 Your use-case was for debugging. Surgically targeting specific parsers isn't needed in that case. Query computation isn't that expensive and clearing the cache isn't that frequent.